app/main.py

```diff
@@ -1,3 +1,44 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    caching_result = {}
+    wraps(func)
```

Comment:
You need to apply `wraps` as a decorator for `wrapper`, you can see an example [in the documentation](https://docs.python.org/3/library/functools.html#functools.wraps).